aws/auto-scaling-group/vpc.tf

```diff
@@ -106,23 +134,64 @@ resource "aws_route_table" "rt" {
   }
 
   tags = merge(var.tags, {
-    Name = "${var.name}-rt"
+    Name = "${var.name}-rt_public_subnets"
   })
 }
 
-# A route table association should be performed.
-# Otherwise, AWS creates another default route table for the VPC,
-# and the subnets do not get automatically associated to the correct route table,
-# which would mean that internet traffic would not be re-routed to the internet
-# gateway.
-resource "aws_main_route_table_association" "main_rt" {
-  vpc_id         = aws_vpc.vpc.id
-  route_table_id = aws_route_table.rt.id
+resource "aws_route_table_association" "rta_private" {
+  count = length(aws_subnet.instances_subnets)
+
+  # If you do not choose a multi-AZ NAT Gateway deployment, there is only
+  # one single NAT Gateway, so there is only one single route table routing
+  # traffic to the single NAT Gateway.
+  route_table_id = var.multi_az_nat ? aws_route_table.rt_private_subnets[count.index].id : aws_route_table.rt_private_subnets[0].id
```

Comment:
That's a great suggestion, I think that you are missing the `aws_route_table` resource `id` field in your suggestion though. 
I changed the line of code to:

```tf
route_table_id = aws_route_table.rt_private_subnets[min(count.index, length(aws_route_table.rt_private_subnets) - 1)].id
```

I will also keep the comment explaining this line of code, because it might make understanding the motivation behind it easier for other devs (or me in the future 💩 )